File: frontend/src/domain/payment.ts
export type PaymentStatus = "PENDING" | "SUCCESS" | "FAILED";

export interface Payment {
  id: string;
  reservation_id: string;
  provider: string;
  amount: number;
  status: PaymentStatus;
  transaction_id?: string | null;
  created_at: string;
}

File: backend/core/domain/model/payment.go
package model

import "time"

type PaymentStatus string

const (
	PaymentPending PaymentStatus = "PENDING"
	PaymentSuccess PaymentStatus = "SUCCESS"
	PaymentFailed  PaymentStatus = "FAILED"
)

type Payment struct {
	ID            string        `gorm:"type:uuid;default:uuid_generate_v4();primaryKey"`
	ReservationID string        `gorm:"type:uuid;not null"`
	Provider      string        `gorm:"size:50;not null"`
	Amount        int           `gorm:"not null"`
	Status        PaymentStatus `gorm:"type:payment_status;not null"`
	TransactionID *string       `gorm:"size:100"`
	CreatedAt     time.Time     `gorm:"autoCreateTime"`

	Reservation Reservation `gorm:"foreignKey:ReservationID"`
}

func (Payment) TableName() string {
	return "payments"
}
